### DISCUSSION TITLE
Feature to run nodejs modules in Run Script

### DISCUSSION THREAD

**Discussion by sondk** [synfron/ReshaperForBurp#73] [Ideas]
Hello,
This is an awesome extension with many custom rules, especially **Run Script**.
However, when I want to import big node module, for example `https://cdn.jsdelivr.net/npm/node-forge@1.0.0/dist/forge.min.js` in **Run Script**, it return `Fail running rule`.
Could you add a new feature so we are able to import node modules from URL or file please!

**Comment by ddwightx** [comment]
The best way to use a JS library like this is to copy the script text in a global variable, then to `eval ` the script like so:
`eval(Reshaper.variables.getGlobalVariable("Node-Forge"));`

Unfortunately, there are incompatible with the `console` object Reshaper provides. The easy way to overcome this is to set `console` to `undefined` like so:

```
let consoleBackup = console;
console = undefined;
eval(Reshaper.variables.getGlobalVariable("Node-Forge"));
console = consoleBackup;
console.log("hello world");
```

**Reply by ddwightx** [reply]
@sondk I have edited the answer above with a better working solution (I hope. Haven't tried the libraries functionality itself)

**Reply by sondk** [reply]
Could I fix it by updating JavaScript core engine of Reshaper?

**Reply by ddwightx** [reply]
I don't believe there is anything left to fix given the solution above (unless you're seeing other errors).

**Reply by sondk** [reply]
it's working now. Thank you so much!
